### ISSUE TITLE
Subscription output definition does not work

### ISSUE THREAD

**Issue by LDTips** [openconfig/gnmic#278]
When binding subscriptions to outputs like shown in the [relevant documentation](https://gnmic.openconfig.net/user_guide/subscriptions/#binding-subscriptions), the subscriptions send the data to all outputs instead only to the ones defined for each subscription.

For example in the configuration below, port_stats should be sent only to output1 and route_stats only to output2. Instead, the observed behavior is both port_stats and route_stats being sent to two outputs instead of the respective ones. Am I missing some configuration key, or is it a bug?

```yaml
outputs:
  output1:
    type: kafka
    name: output1
    # Comma separated brokers addresse
    address: "10.27.250.4:9092"
    # Kafka topic name
    topic: Telemetry-counters
    max-retry: 2
    # Kafka connection timeout
    timeout: 2s
    # Wait time to reestablish the kafka producer connection after a failure
    recovery-wait-time: 5s
    # Exported msg format, json, protojson, prototext, proto, event
    format: event
    split-events: false
    # boolean, if true the message timestamp is changed to current time
    override-timestamps: false
    # Number of kafka producers to be created
    num-workers: 1
    # (bool) enable debug
    debug: false
    # (int) number of messages to buffer before being picked up by the workers
    buffer-size: 0
    # (string) enables compression of produced message. One of gzip, snappy, zstd, lz4
    compression-codec: gzip
    # (bool) enables the collection and export (via prometheus) of output specific metrics
    enable-metrics: false
  output2:
    type: kafka
    name: output2
    # Comma separated brokers addresses
    address: "10.27.250.4:9092"
    # Kafka topic name
    topic: Telemetry-onChange
    max-retry: 2
    # Kafka connection timeout
    timeout: 2s
    # Wait time to reestablish the kafka producer connection after a failure
    recovery-wait-time: 5s
    # Exported msg format, json, protojson, prototext, proto, event
    format: event
    split-events: false
    # boolean, if true the message timestamp is changed to current time
    override-timestamps: false
    # Number of kafka producers to be created
    num-workers: 1
    # (bool) enable debug
    debug: false
    # (int) number of messages to buffer before being picked up by the workers
    buffer-size: 0
    # (string) enables compression of produced message. One of gzip, snappy, zstd, lz4
    compression-codec: gzip
    # (bool) enables the collection and export (via prometheus) of output specific metrics
    enable-metrics: false
 
# gNMI target address; CLI flag `--address`
#address: "192.168.122.102:57400"
# gNMI target user name; CLI flag `--username`
username: admin
# gNMI target user password; CLI flag `--password`
password: admin
# connection mode; CLI flag `--insecure`
#insecure: true
skip-verify: true
# log file location; CLI flag `--log-file`
#log-file: /rpc/gnmic.log
# Enable or disable debugging
debug : false
# Subscription for statistics on a port
subscriptions:  # container for subscriptions
  port_stats:     # a named subscription, a key is a name
    paths:      # list of subscription paths for that named subscription
      - "/state/port/statistics/out-octets"
      - "/state/port/statistics/in-octets"
    stream-mode: sample # one of [on-change target-defined sample]
    sample-interval: 60s
    encoding: bytes
    outputs:
      - output1
  route_stats:
    paths:
      - "/state/port/oper-state"
    stream-mode: on_change
    encoding: json
    outputs:
      - output2
```

**Comment by karimra** [comment]
Which gNMIc version are you using? That functionality is available starting with version 0.32.0.
Did you configure any subscriptions/outputs under `targets` ?
Can you share the log output to make sure the config is properly read ?

**Comment by LDTips** [comment]
Thank you for your help. Indeed it was a version issue. Docker container was used from the wrong repository (not maintained any more), so the version was older. Newer version works properly :)
